Tools/jit/_targets.py

```diff
@@ -537,7 +537,14 @@ def get_target(host: str) -> _COFF | _ELF | _MachO:
         args = ["-fms-runtime-lib=dll"]
         target = _COFF(host, args=args)
     elif re.fullmatch(r"x86_64-.*-linux-gnu", host):
-        args = ["-fno-pic", "-mcmodel=medium", "-mlarge-data-threshold=0"]
+        args = [
+            # Jump tables generate R_X86_64_32S relocations, which assume a
+            # signed 32-bit address space:
+            "-fno-jump-tables",
+            "-fno-pic",
+            "-mcmodel=medium",
+            "-mlarge-data-threshold=0",
+        ]
```

Comment:
With your latest change to use a lookup table in `LOAD_COMMON_CONSTANT`, this isn't needed anymore:

```suggestion
        args = ["-fno-pic", "-mcmodel=medium", "-mlarge-data-threshold=0"]
```